src/common/conn/server.rs

```diff
@@ -381,11 +381,18 @@ where
                 .await
                 .map_or(Err(ServerError::ServerConnectionTimeout), |r| r);
 
-            if connection.is_err() {
-                let _ = self.app_client.take();
-                continue;
-            }
-            if let Err(e) = match connection.unwrap() {
+            let connection = match connection {
+                Ok(c) => c,
+                Err(ServerError::ServerWebRTCError(_)) => {
```

Comment:
I think a comment here is a good idea to explain both why `ServerWebRTCError` is special cased to not destroy the `app_client`, and why the `app_client` gets destroyed for other errors, as I think these error cases form an important part of the state machine.